graylog2-web-interface/src/components/configurationforms/ConfigurationForm.jsx

```diff
@@ -83,7 +83,7 @@ const ConfigurationForm = createReactClass({
   },
 
   _sortByOptionality(x1, x2) {
-    return (this.state.configFields[x1].is_optional - this.state.configFields[x2].is_optional);
+    return this.state.configFields[x1.key].is_optional - this.state.configFields[x2.key].is_optional || x1.pos - x2.pos;
```

Comment:
I think we should rewrite this in a more explicit way.
Subtracting two booleans is already hard to understand without deep understanding of JS internals, and using the `or` operator to do another comparison when the previous result was 0 only makes it harder to understand.